node/lib/util/commit.js

```diff
@@ -301,23 +362,26 @@ exports.commit = co.wrap(function *(metaRepo, all, metaStatus, message) {
     const subCommitters = Object.keys(submodules).map(commitSubmodule);
```

Comment:
I don't think anyone will fault you for finding something in JavaScript to be ugly :). The library you link is callback-based, which is a couple of generations behind what we're doing. First there was the node-style callbacks used in the `async` library you reference, which was functional but led to callback hell. Promises were next and made things much better but were still very verbose and still required lots of closures. The generators-approach lets you write async code that looks very similar to synch code (i.e., not many callbacks). Admittedly though, any "idioms" such as `yield someArray` to achieve parallelism are pretty new.